### PR TITLE
fix(benchmark): resolve PPO worker isolation deadlock and step timeouts

### PR DESCRIPTION
**Pull request by ll7**
## What/Why

This PR resolves the PPO worker isolation deadlock and lazy-loading step timeouts (issue #5498). 

### The Problem
1. **Deadlock**: The PyTorch model loading in `PPOPlanner.__init__` was occurring in the parent process. Because the runner uses `fork` multiprocessing to isolate planner execution, the background PyTorch/OpenMP threads initialized in the parent process deadlocked inside the child process worker due to lock copying.
2. **Step Timeout**: Delaying the model initialization to `step()` meant the very first planner step had to download and load the model. Since `POLICY_STEP_TIMEOUT_SECS` is set to a strict `0.2` seconds, this first load always timed out, causing the worker process to be terminated, re-spawned, and timed out in a loop, degrading the execution to zero-velocity fallbacks.

### The Fix
1. **Lazy Loading**: Deferred `PPOPlanner` model loading (`_load_model` and `_init_predictive_foresight`) to an explicit `_ensure_model_loaded()` helper.
2. **Startup Handshake**: Implemented a worker initialization handshake in `_ensure_worker` (`robot_sf/benchmark/runner.py`) and `_planner_step_worker`. The child process worker calls `_ensure_model_loaded()` immediately upon startup (before entering the step command loop) and sends an `init_ok` handshake back to the parent.
3. **No step-timer penalty**: The parent blocks waiting for the handshake before allowing any step requests. This shifts the model loading overhead out of the `step()` execution timer window, ensuring it doesn't count against the `0.2`s step budget.
4. **Thread safety**: Configured PyTorch thread count via `torch.set_num_threads(1)` inside both the planner constructor and the isolated worker child process to minimize thread pool overhead.

## Validation

1. **CPU Target Execution**: Successfully executed `classic_cross_trap_high--111` target (which uses PPO). It ran natively in ~5 seconds with no fallback/degraded actions recorded, producing bitwise identical hashes across all 3 repeats.
2. **Focused Tests**: Added two unit tests (`test_planner_step_process_handshake_and_lazy_loading` and `test_planner_step_process_handshake_failure_propagation`) to verify the worker handshake and startup error propagation.
3. **Pytest Run**: All 49 tests passed successfully in 8.8s:
   ```bash
   uv run pytest tests/benchmark/test_exact_repeat_executor.py tests/benchmark/test_exact_repeat_campaign.py
   ```
4. **Style**: Lint and format checks pass cleanly:
   ```bash
   uv run ruff check robot_sf/baselines/ppo.py robot_sf/benchmark/runner.py tests/benchmark/test_exact_repeat_executor.py
   uv run ruff format robot_sf/baselines/ppo.py robot_sf/benchmark/runner.py tests/benchmark/test_exact_repeat_executor.py --check
   ```

## PR Metadata

This PR was produced by the agy/Gemini-3.5-Flash cheap implementation lane; the review gate hardens and merges.

Refs #5498. successor slice: PPO worker isolation and lazy-loading step timeouts; does not duplicate PR #5499.

### Artifact / output handling
`output/`: NA — no campaign, no SLURM, no training, no benchmark artifacts produced; the only generated outputs are local target cache files for CPU validation. Nothing to promote to durable storage. Downstream propagation: NA (bugfix slice, no research claim).

### CODE REVIEW

**Comment by ll7** [comment]
## Gate findings — PR #5740 @ `a456f79c4`

Phase 1 reviewed the complete 92-line runtime/test diff against `origin/main` (`c0acb8e7d45353145b12dfbc28a60af2f9a65fde`) and linked campaign issue #5498. The worker-handshake direction was in scope, but the initial head was not mergeable under the fail-closed gate.

1. HIGH — Established PPO error policy regressed: moving `_load_model()` out of `PPOPlanner.__init__` meant missing or incompatible models with `fallback_to_goal=False` no longer raised the required remediation error at construction. Hosted `fast-feedback (3)` failed both `tests/test_error_policy.py` cases. Verdict: fixed in `ac4eb638b` by restoring eager fail-closed loading for direct construction and making only the benchmark PPO loader explicitly defer loading into the fork worker.

2. MEDIUM — Broad-exception ratchet failure: the initial worker startup paths added unapproved broad handlers at `robot_sf/benchmark/runner.py:301`, `:304`, and `:417`; hosted `fast-feedback (1)` failed the broad-exception check. Verdict: fixed in `ac4eb638b` by narrowing the IPC catches and registering the single intentional child-init catch in `scripts/validation/broad_exception_baseline.json`.

**Open review threads: 0 open review threads.** CodeRabbit supplied only a rate-limit comment; no CodeRabbit or other inline review thread is open. The bot comment is advisory and does not replace this gate review.

Re-review delta at `a456f79c4`: `gh pr update-branch` merged current `main`, adding the already-reviewed #5741 seed/fixture slice; no new finding was introduced and the prior repairs stand. Hosted CI for the updated head, staleness, and the final merge checklist must still be green.

**Comment by ll7** [comment]
Gate delta: prior findings review at `a456f79c46f7e63d5207bbb31b762b9780df577a` stands; unchanged head, so no diff re-review was performed. Resuming phase 2/3 verification and merge checklist.

**Comment by ll7** [comment]
Re-review delta at `28b72214b01d6b0438b8bdeae10e895986818280`: `gh pr update-branch` added only the current-main merge delta (lease/reaper files and the chance-constrained provider test), disjoint from the PR’s five changed files. Prior findings review at `a456f79c46f7e63d5207bbb31b762b9780df577a` still stands; no new finding.

**Comment by ll7** [comment]
## Gate findings update — PR #5740 @ `317616999`

1. MEDIUM — The new Torch `ImportError` handlers expanded the repository optional-import inventory from 76 to 78 without updating the guard contract, causing the full readiness gate to fail in `test_optional_import_guard_inventory.py`. Verdict: fixed in `317616999` by using the canonical `robot_sf.common.optional_import.try_import()` helper in `PPOPlanner` and the isolated worker; the snapshot remains unchanged except for required line-number samples.

**Open review threads: 0 open review threads.** No CodeRabbit, Gemini, or other inline review-bot thread is open. The prior gate findings at `a456f79c4` remain fixed.

**Comment by ll7** [comment]
Gate delta at `87800e310a81b20389a6f468f9f94917695fb366`: the PR branch is now up to date with current `main` (`3ef3915f`). The merge delta is limited to current-main changes and is disjoint from the PPO worker/error-policy files covered by the prior findings; prior findings remain fixed. Focused regression and the full configured readiness rerun remain part of the merge checklist.

**0 open review threads.** No new review-bot thread was introduced by the branch refresh.

**Comment by ll7** [comment]
## Gate findings update — PR #5740 @ `71c8ebc4505365656246cdcdb141616b1b2cca17`

1. **MEDIUM gate finding** — clean configured readiness reported changed-file coverage below the 80% minimum for `robot_sf/benchmark/runner.py` (37.5%, 12/32 changed executable lines). **Verdict: fixed** by adding direct coverage for the PPO deferred-loader branch, worker initialization/step/error paths, and parent handshake timeout/receive-error paths in `tests/unit/test_runner_helper_coverage.py` (`71c8ebc45`).

The corrected focused validation passes 52 tests with Ruff clean. **0 open review threads.** The prior review-bot findings remain fixed; no review-bot thread was introduced by this change.

**Comment by ll7** [comment]
Gate delta at `4a8eac0994de0948fd268611a5e2d927ab323fc5`: the branch was refreshed against post-#5745 current `main` (`250c661a`). The merge delta includes the already-gated #5745 calibration merge and current docs-proof consistency helper/tests, all outside #5740's PPO worker/error-policy files. Prior findings and the 100% changed-runner coverage proof remain valid.

**0 open review threads.** No new review-bot thread was introduced by the refresh.
